text-extraction/src/main/java/edu/kit/kastel/mcse/ardoco/core/textextraction/NounMappingImpl.java

```diff
@@ -316,52 +217,41 @@ public boolean equals(Object obj) {
         return Objects.equals(getReference(), other.getReference());
     }
 
-    /**
-     * @return if this is a phrase or contains a phrase
-     */
     @Override
-    public boolean isPhrase() {
-        return hasPhrase;
+    public boolean isTheSameAs(NounMapping other) {
```

Comment:
isTheSameAs should be equals()

Reply:
Or don't call it isTheSameAs .. e.g., hasSameReference(NM other)

Reply:
And we should maybe drop every "equals" in Noun and Phrase Mapping as there are not "the one equals"